tests/beamlines/unit_tests/test_i24.py

```diff
@@ -1,15 +1,17 @@
 import sys
+from importlib import reload
 from unittest.mock import patch
 
+import dodal.utils
+from dodal.beamlines import beamline_utils, i24
 from dodal.devices.i24.i24_vgonio import VGonio
+from dodal.utils import make_all_devices
 
-with patch.dict("os.environ", {"BEAMLINE": "i24"}, clear=True):
-    from dodal.beamlines import beamline_utils, i24
-    from dodal.utils import make_all_devices
 
-
-def test_device_creation():
-    devices = make_all_devices(i24, fake_with_ophyd_sim=True)
+@patch.object(dodal.utils, "get_beamline_name", return_value="i24")
```

Comment:
I prefer the solution in https://github.com/DiamondLightSource/dodal/pull/330#pullrequestreview-1877164642 - I know it's just a test, but reloading modules and deleting from sys feels extra messy to me - and it's not really necessary - all that's needed is to set the `beamline_utils.BL` and clear the devices like I did here https://github.com/DiamondLightSource/dodal/commit/05ab615825f85caef0d1cdb64ef7d3f512483e40

Reply:
Now that 330 is merged I'm happy to just rebase on top of main again and discard this. Any thoughts on the rest of the changes?

Reply:
Yes, I think this is great! it'll be very helpful I'm sure. It looks detailed enough for most developers to me, hopefully that's right. But I think you'd get better feedback from someone who didn't write the thing in the first place, haha

Reply:
Removed the i24 test changes from the tree

Reply:
But now i24 is failing again

Reply:
I'm looking into it, I think it's not relevant for this PR, you can merge it anyway

Reply:
https://github.com/DiamondLightSource/dodal/issues/331